### PR TITLE
Fix an infinite loop issue when playing media.

### PR DESCRIPTION
**Pull request by Sunwoo-Nam**
Browser process has to share hardware latency with renderer process
for audio streaming. In castanets case, browser process should sends
an audio buffer including the latency when changing it.
If it doesn't, renderer process has a latency as garbage value and
it can occur infinite loop.

### CODE REVIEW

File: media/audio/audio_device_thread.cc [hunk 1]
```diff
@@ -98,18 +98,29 @@ void AudioDeviceThread::ThreadMain() {
...
 #if defined(CASTANETS)
 #if defined(OS_WIN)
     size_t bytes_read = 0;
 #else
+#if !defined(NETWORK_SHARED_MEMORY)
+    // Receive AudioOutputBuffer data to know delay time.
+    size_t buffer_bytes_read = HANDLE_EINTR(recv(
+         client_handle_.get().handle, buffer_data, buffer_size, MSG_WAITALL));
+    if (buffer_bytes_read != buffer_size)
+      break;
```
Comment (Chikdolman) [inline media/audio/audio_device_thread.cc:117]: What happens to the last part of the data? will the last part of data's buffer_bytes_read == buffer_size?
if not, doesn't it skip the last data by not copying to the shared_memory?

**Review by Chikdolman — COMMENTED** [review]
I have a question. Please check.

File: media/audio/audio_device_thread.cc (hunk 1 above)
Comment (Sunwoo-Nam) [reply media/audio/audio_device_thread.cc:117]: buffer_size is a size of shared memory. shared memory size is fixed initially.
BP sends a buffer and RP receives it with the same size.

**Review by Chikdolman — APPROVED** [review]
LGTM. This patch must be included in order to load Youtube page.

**Pull request merged**